### ISSUE TITLE
exception in scons while using m to install ninja

### ISSUE THREAD

**Issue by IanWhalen** [mongodb/server-workflow-tool#6]
```
> m setup-dev-env.macos-extra

Installing ninja
brew install ninja
==> Downloading https://homebrew.bintray.com/bottles/ninja-1.9.0.mojave.bottle.tar.gz
==> Pouring ninja-1.9.0.mojave.bottle.tar.gz
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d

zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
==> Summary
🍺  /usr/local/Cellar/ninja/1.9.0: 7 files, 275.8KB
mkdir -p /Users/ian/mongo/src/mongo/db/modules
cd /Users/ian/mongo/src/mongo/db/modules && git clone https://github.com/RedBeard0531/mongo_module_ninja ninja
Cloning into 'ninja'...
cd /Users/ian/mongo && python buildscripts/scons.py CC=clang CXX=clang++ CCFLAGS=-Wa,--compress-debug-sections MONGO_VERSION='0.0.0' MONGO_GIT_HASH='unknown' VARIANT_DIR=ninja --modules=ninja build.ninja
scons: Reading SConscript files ...
  File "/Users/ian/mongo/SConstruct", line 933

    print( "python version: " + " ".join( [ `i` for i in sys.version_info ] ) )

                                            ^

SyntaxError: invalid syntax```

**Comment by IanWhalen** [comment]
FWIW I'm continuing to debug in case this is somehow related to my venv setup.

Looks like backticks were deprecated in Python 3, so this script instead needs to run with Python 2?

**Comment by visemet** [comment]
> Looks like backticks were deprecated in Python 3, so this script instead needs to run with Python 2?

Until [SERVER-32295](https://jira.mongodb.org/browse/SERVER-32295) goes through, SCons needs to be run with Python 2. Afterwards it'll need to be run with Python 3.

**Comment by milkie** [comment]
That ticket implies that afterwards, scons could be run with Python 3, not that it would be required to run with Python 3.  Which is it?

**Comment by visemet** [comment]
> That ticket implies that afterwards, scons could be run with Python 3, not that it would be required to run with Python 3. Which is it?

The Dev Tools team isn't attempting to support running SCons or any of our `buildscripts/` with both Python 2 and Python 3. My understanding is that there is going to be a hard cutover to Python 3 in Evergreen and for your local environment.

**Comment by milkie** [comment]
So that would be a hard cutover for all [supported] branches at the same time?

**Comment by visemet** [comment]
> So that would be a hard cutover for all [supported] branches at the same time?

I think only the master branch (and the v4.2 branch that'll eventually be created) will use Python 3. The older stable branches will continue to use Python 2.

**Comment by milkie** [comment]
Okay, so in summary, we'll need the new-engineer setup script to ensure that both Python 2 and Python 3 are installed, and to set up shortcuts such that python scripts are invoked using a #! in the script itself.. is that correct?
Otherwise, shortcuts that hardcode running 'python' (or python3 or python2) won't work when switching between branches.

**Comment by IanWhalen** [comment]
So far I've just been hardcoding py2/py3 to make this work for master
alone, and then we can fix it up once the py3 changes land.

**Comment by visemet** [comment]
> Okay, so in summary, we'll need the new-engineer setup script to ensure that both Python 2 and Python 3 are installed, and to set up shortcuts such that python scripts are invoked using a #! in the script itself.. is that correct?

Is it a requirement for this tool to work with all supported branches? If so, I think shebangs are going to be the path of least resistance. Do we know if it is possible to use a single virtual environment (well one for Python 2 and another for Python 3) across all supported branches?

**Comment by IanWhalen** [comment]
Fixed in fd3f596fef52c85d20679eb3cf55b4dc2172e96d
